frontend/src/routes/Login.jsx

```diff
@@ -1,50 +1,88 @@
-import { useNavigate } from "react-router-dom";
-import { useState } from "react";
+import { useNavigate } from 'react-router-dom';
+import { useState } from 'react';
 import { GoogleLogin } from '@react-oauth/google';
-import "../styles/Login.css";
-import useAuth from "../hooks/useAuth";
+import '../styles/Login.css';
+import useAuth from '../hooks/useAuth';
+import { jwtDecode } from 'jwt-decode';
+import { serverFetch } from '../hooks/serverUtils';
 
 function LoginForm() {
-  const [loginForm, setLoginForm] = useState({ username: "", password: "" });
+  const [loginForm, setLoginForm] = useState({ username: '', password: '' });
   const navigate = useNavigate();
-  const { signIn, signInWithGoogle } = useAuth()
+  const { signIn, signInWithGoogle } = useAuth();
 
   function onChange(event) {
     const { name, value } = event.target;
     setLoginForm((oldForm) => ({ ...oldForm, [name]: value }));
   }
 
-  function onSubmit(e) {
-    e.preventDefault();
-
+  async function loginUser(hash) {
     const data = {
       username: loginForm.username,
-      password: loginForm.password,
+      hashedPassword: hash
     };
 
+    const endpoint = '/login/login';
     const options = {
-      method: "POST",
+      method: 'POST',
       headers: {
-        "Content-Type": "application/json",
+        'Content-Type': 'application/json'
       },
-      body: JSON.stringify(data),
+      body: JSON.stringify(data)
     };
-    fetch("/login/login", options);
+
+    try {
+      const response = await serverFetch(endpoint, options);
+      if (response.ok) {
+        const data = await response.json();
+        signIn();
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  async function onSubmit(e) {
+    e.preventDefault();
+
+    try {
+      const saltRounds = 10;
+      const salt = await bcrypt.genSalt(saltRounds);
+      const hash = await bcrypt.hash(loginForm.password, salt);
```

Comment:
We could extract this to a function as it this is repeated in 2 places if I'm not mistaken.